providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -0,0 +1,203 @@
+        this.stateConsumer = stateConsumer;
+        jsonLogicHandler = new JsonLogic();
+    }
+
+    /**
+     * Initialize in-process resolver.
+     */
+    public void init() throws Exception {
+        flagStore.init();
+        final Thread stateWatcher = new Thread(() -> {
+            try {
+                while (true) {
+                    final StorageState storageState = flagStore.getStateQueue().take();
+                    switch (storageState) {
+                        case OK:
+                            stateConsumer.accept(ProviderState.READY);
+                            break;
+                        case ERROR:
+                            stateConsumer.accept(ProviderState.ERROR);
+                            break;
+                        case STALE:
+                            // todo set stale state
+                        default:
+                            log.log(Level.INFO, String.format("Storage emitted unknown status: %s", storageState));
+                    }
+                }
+            } catch (InterruptedException e) {
+                log.log(Level.WARNING, "Storage state watcher interrupted", e);
+            }
+        });
+        stateWatcher.setDaemon(true);
+        stateWatcher.start();
+    }
+
+    /**
+     * Shutdown in-process resolver.
+     */
+    public void shutdown() {
+        flagStore.shutdown();
+    }
+
+    /**
+     * Resolve a boolean flag.
+     */
+    public ProviderEvaluation<Boolean> booleanEvaluation(String key, Boolean defaultValue,
+                                                         EvaluationContext ctx) {
+        return resolveGeneric(Boolean.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve a string flag.
+     */
+    public ProviderEvaluation<String> stringEvaluation(String key, String defaultValue,
+                                                       EvaluationContext ctx) {
+        return resolveGeneric(String.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve a double flag.
+     */
+    public ProviderEvaluation<Double> doubleEvaluation(String key, Double defaultValue,
+                                                       EvaluationContext ctx) {
+        return resolveGeneric(Double.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve an integer flag.
+     */
+    public ProviderEvaluation<Integer> integerEvaluation(String key, Integer defaultValue,
+                                                         EvaluationContext ctx) {
+        return resolveGeneric(Integer.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve an object flag.
+     */
+    public ProviderEvaluation<Value> objectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        return resolveGeneric(Value.class, key, defaultValue, ctx);
+    }
+
+    private <T> ProviderEvaluation<T> resolveGeneric(Class<T> type, String key, T defaultValue,
```

Comment:
```suggestion
    private <T> ProviderEvaluation<T> resolve(Class<T> type, String key, T defaultValue,
```

nit, but I think the fact this is generic is communicated by the signature.

Reply:
Done with the latest commit